File: Experiment Processing/recommender/distance_metrics/chebyshev_distance.py
from recommender.distance_metrics.abstract_distance_metric import AbstractDistanceMetric


class ChebyshevDistance(AbstractDistanceMetric):
    """
    Chebyshev distance, see https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.distance.chebyshev.html
    for more information.
    """

    def get_name(self):
        return "Chebyshev Distance"

    def calculate_distance(self, track_a, track_b):
        max_distance = 0

        track_a_features = track_a.get_metadata()
        track_b_features = track_b.get_metadata()

        for attribute in self.attribute_keys:
            attribute_own = track_a_features[attribute]
            attribute_other = track_b_features[attribute]

            temp_distance = abs(attribute_own - attribute_other)
            if temp_distance > max_distance:
                max_distance = temp_distance

        return max_distance

File: Web Survey/resources/js/recommender/distance_metrics/ChebyshevDistance.js
import AbstractDistanceMetric from "./AbstractDistanceMetric";

export default class ChebyshevDistance extends AbstractDistanceMetric {

    static getMetricName() {
        return "Chebyshev Distance";
    }

    calculateDistance(track, otherTrack, attributes) {
        const self = this;

        let maxDistance = 0;

        this.attributeKeys.forEach(attribute => {

            let attributeOwn;
            let attributeOther;

            if(attribute === "loudness") {
                attributeOwn = self.fixLoudness(attributes[track][attribute]);
                attributeOther = self.fixLoudness(attributes[otherTrack][attribute])
            } else {
                attributeOwn = attributes[track][attribute];
                attributeOther = attributes[otherTrack][attribute];
            }

            const tempDistance = Math.abs(attributeOwn - attributeOther);
            if(tempDistance > maxDistance) {
                maxDistance = tempDistance;
            }
        });

        return maxDistance;
    }


}